lib/ansible/module_utils/redfish_utils.py

```diff
@@ -0,0 +1,932 @@
+        data = response['data']
+
+        processors_uri = data[key]["@odata.id"]
+
+        # Get a list of all CPUs and build respective URIs
+        response = self.get_request(self.root_uri + processors_uri)
+        if response['ret'] is False:
+            return response
+        result['ret'] = True
+        data = response['data']
+
+        for cpu in data[u'Members']:
+            cpu_list.append(cpu[u'@odata.id'])
+        for c in cpu_list:
+            cpu = {}
+            uri = self.root_uri + c
+            response = self.get_request(uri)
+            if response['ret'] is False:
+                return response
+            data = response['data']
+
+            cpu['Name'] = data[u'Id']
+            cpu['Manufacturer'] = data[u'Manufacturer']
+            cpu['Model'] = data[u'Model']
+            cpu['MaxSpeedMHz'] = data[u'MaxSpeedMHz']
+            cpu['TotalCores'] = data[u'TotalCores']
+            cpu['TotalThreads'] = data[u'TotalThreads']
+            cpu['State'] = data[u'Status'][u'State']
+            cpu['Health'] = data[u'Status'][u'Health']
+            cpu_details.append(cpu)
+        result["entries"] = cpu_details
+        return result
+
+    def get_nic_inventory(self):
+        result = {}
+        nic_details = []
+        nic_list = []
+        key = "EthernetInterfaces"
+
+        # Search for 'key' entry and extract URI from it
+        response = self.get_request(self.root_uri + self.systems_uri)
+        if response['ret'] is False:
+            return response
+        result['ret'] = True
+        data = response['data']
+
+        ethernetinterfaces_uri = data[key]["@odata.id"]
+
+        # Get a list of all network controllers and build respective URIs
+        response = self.get_request(self.root_uri + ethernetinterfaces_uri)
+        if response['ret'] is False:
+            return response
+        result['ret'] = True
+        data = response['data']
+
+        for nic in data[u'Members']:
+            nic_list.append(nic[u'@odata.id'])
+
+        for n in nic_list:
+            nic = {}
+            uri = self.root_uri + n
+            response = self.get_request(uri)
+            if response['ret'] is False:
+                return response
+            data = response['data']
+
+            nic['Name'] = data[u'Name']
+            nic['FQDN'] = data[u'FQDN']
+            for d in data[u'IPv4Addresses']:
+                nic['IPv4'] = d[u'Address']
+                if 'GateWay' in d:   # not always available
+                    nic['Gateway'] = d[u'GateWay']
+                nic['SubnetMask'] = d[u'SubnetMask']
+            for d in data[u'IPv6Addresses']:
+                nic['IPv6'] = d[u'Address']
+            for d in data[u'NameServers']:
+                nic['NameServers'] = d
+            nic['MACAddress'] = data[u'PermanentMACAddress']
+            nic['SpeedMbps'] = data[u'SpeedMbps']
+            nic['MTU'] = data[u'MTUSize']
+            nic['AutoNeg'] = data[u'AutoNeg']
```

Comment:
Since all of the properties above are optional, directly accessing them might not be the best idea. One way of handling this is to use `dict.get()` method that will return `None` if the key is not present. The other option would be to only access the value if the key is present.